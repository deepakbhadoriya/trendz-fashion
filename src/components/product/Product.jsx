import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import cartAction from '../../redux/actions/cartAction';

import HeartIcon from '../../assets/icons/heart/heart';
import PlusIcon from '../../assets/icons/plus/plus';

import styles from './Product.module.scss';

const Product = ({ product }) => {
  const dispatch = useDispatch();
  const cart = useSelector((state) => state.cart);

  return (
    <div className="col-lg-3 col-md-4 col-sm-6 col-12 my-3 ">
      <div className={styles.product} key={product.id} tabIndex="0">
        <Link to={`/product/${product.id}`} className={styles.productLink}>
          <div
            className={styles.productImage}
            style={{ backgroundImage: `url('${product.image}')` }}
            alt={product.title}
          />
          <div className={styles.productDetails}>
            <h2 className={styles.productTitle} title={product.title}>
              {product.title}
            </h2>
            <div className={styles.price}>
              <span className={styles.productPrice}>
                {product.currencySymbol || '$'} {product.price}
              </span>
            </div>
          </div>
        </Link>
        <div className={styles.actions}>
          <button
            className={styles.addToCart}
            onClick={() => dispatch(cartAction.addToCart(product))}
          >
            <PlusIcon />
          </button>
          {/* Only if user logged in */}
          <button className={styles.addToWishlist}>
            <HeartIcon />
          </button>
        </div>
      </div>
    </div>
  );
};

export default Product;
