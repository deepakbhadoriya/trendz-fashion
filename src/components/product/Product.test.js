import React from 'react';

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';

import Product from './Product';
import TestEnvironmentWrapper from '../../test/TestEnvironmentWrapper';

const { mount } = Enzyme;
Enzyme.configure({ adapter: new Adapter() });

describe('<Product/>', () => {
  it('render correctly (snapshot)', () => {
    const tree = mount(
      <TestEnvironmentWrapper>
        <Product product={{ id: 5 }} />
      </TestEnvironmentWrapper>
    );
    expect(toJson(tree)).toMatchSnapshot();
  });
});
