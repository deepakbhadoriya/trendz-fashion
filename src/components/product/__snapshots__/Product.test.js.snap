// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Product/> render correctly (snapshot) 1`] = `
<TestEnvironmentWrapper>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Provider
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
      >
        <IntlProvider
          defaultFormats={Object {}}
          defaultLocale="en"
          formats={Object {}}
          locale="en"
          messages={
            Object {
              "electronics": "Electronics",
              "fifty_percent_off": "Get up to 50%",
              "help_center": "Help Center",
              "holiday_season": "For the holiday season",
              "jewellery": "Jewellery",
              "login": "Log in",
              "mensClothing": "Men's clothing",
              "referral_message": "Invite a friend and get a discount of 5%",
              "register": "Register",
              "search": "Search for products",
              "thirty_percent_discount_shirts": "Get up to 30% off shirts",
              "womenClothing": "Women's Clothing",
            }
          }
          onError={[Function]}
          textComponent={Symbol(react.fragment)}
        >
          <Product
            product={
              Object {
                "id": 5,
              }
            }
          >
            <Link
              className="product"
              key="5"
              to="/product/5"
            >
              <LinkAnchor
                className="product"
                href="/product/5"
                navigate={[Function]}
              >
                <a
                  className="product"
                  href="/product/5"
                  onClick={[Function]}
                >
                  <div
                    className="productImage"
                    style={
                      Object {
                        "backgroundImage": "url('undefined')",
                      }
                    }
                  />
                  <div
                    className="productDetails"
                  >
                    <h2
                      className="productTitle"
                    />
                    <div
                      className="price"
                    >
                      <span
                        className="productPrice"
                      >
                        $
                         
                      </span>
                    </div>
                  </div>
                </a>
              </LinkAnchor>
            </Link>
          </Product>
        </IntlProvider>
      </Provider>
    </Router>
  </BrowserRouter>
</TestEnvironmentWrapper>
`;
